url8c14fd53ee_luca89_itaco_tgz-pJ8-main_RunnerJ8/src/compilatore/JasminTarget.java

```
@@ -0,0 +1,636 @@
+		/**
+		 * Creo un parser LALR1
+		 */
+		LALR1Generator generator = new LALR1Generator(new ParserSpec());
+		/**
+		 * Prendi la parsing table generata
+		 */
+		LRParsingTable table = generator.getParsingTable();
+		/**
+		 * Crea un parser LR dalla tabella generata
+		 */
+		LRParser parser = new LRParser(table);
+		/**
+		 * Usa il parser per generare l'AST dallo stream di token passati dallo
+		 * scanner
+		 */
+		S result = (S) parser.parse(new Scanner(new FileReader(sorgenteFile)));
+		/**
+		 * Genero il codice Jasmin
+		 */
+		String nomeClasse = sorgenteFile.getName().split("\\.")[0];
+		ByteArrayOutputStream jasminAssemblyBytes = new ByteArrayOutputStream();
+		JasminTarget jt = new JasminTarget(nomeClasse, jasminAssemblyBytes);
+		result.scriviCodice(jt);
+		if (salvaAssembly) {
+			String percorsoFileJ;
+			if (sorgenteFile.getParent() != null) {
+				percorsoFileJ = sorgenteFile.getParent() + File.separator
+						+ nomeClasse + ".j";
+			} else {
+				percorsoFileJ = nomeClasse + ".j";
+			}
+
+			FileOutputStream fileOut = new FileOutputStream(percorsoFileJ);
+			fileOut.write(jasminAssemblyBytes.toByteArray());
+			fileOut.close();
+		}
+		/*
+		 * Genero il bytecode
+		 */
+		String percorsoFileClass;
+		if (sorgenteFile.getParent() != null) {
+			percorsoFileClass = sorgenteFile.getParent() + File.separator
+					+ nomeClasse + ".class";
+		} else {
+			percorsoFileClass = nomeClasse + ".class";
+		}
+		/*
+		 * Oggetto della libreria Jasmin che si occupa di generare il file
+		 * .class
+		 */
+		ClassFile classFile = new ClassFile();
+		/**
+		 * Questa chiamata rocambolesca converte l'output a byte del nostro
+		 * codice Jasmin in input a caratteri necessario per la libreria Jasmin
+		 */
+		CharArrayReader assemblyInput = new CharArrayReader(new String(
+				jasminAssemblyBytes.toByteArray()).toCharArray());
+		try {
+			classFile.readJasmin(assemblyInput, sorgenteFile.getName(), false);
+		} catch (Exception e) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 424.